app/src/main/java/my/noveldokusha/scraper/utils.kt

```diff
@@ -42,14 +42,19 @@ suspend fun connect(url: String): Connection = Jsoup.connect(url).apply {
     ignoreHttpErrors(true)
 }
 
+// pasre if connect success or return empty list
+fun connect.parseOrEmpty(): Document? = if (this.statusCode == 200) this.parse() else listOf()
+
 suspend fun fetchDoc(url: String, timeoutMilliseconds: Int = 20 * 1000): Document = connect(url).timeout(timeoutMilliseconds).getIO()
 suspend fun fetchDoc(url: Uri.Builder, timeoutMilliseconds: Int = 20 * 1000) = fetchDoc(url.toString(), timeoutMilliseconds)
 
 suspend fun <T> tryConnect(extraErrorInfo: String = "", call: suspend () -> Response<T>): Response<T> = try
 {
     call()
-} catch (e: SocketTimeoutException)
-{
+} catch (e: HttpStatusException) {
+	// alternative if ignoreHttpErrors is not implemented
+	Response.success(listOf())
```

Comment:
Wrong, you must detect the are no more pages in the source scraper not here. Usually pages have an index on the bottom to show how many pages of results are, use that or similar.

Reply:
for example suppose there are 10 pages. when opening the 11th page will produce response 404.?
because the repeated calling of the page from is not set from this class isn't it? (index parameter) is obtained from another function.
then how do we see that the page (index + 1) really exists?

Reply:
Same as previous comment.